error_handling/app.errors.js

```diff
@@ -9,12 +9,18 @@ exports.handleCustomError = (err, req, res, next) => {
 };
 
 exports.handlePsqlError = (err, req, res, next) => {
-  if (err.code === "22P02" || err.code === '42601') {
+  if (err.code === "22P02" || err.code === "42601") {
     res.status(400).send({ msg: "Invalid Input" });
   } else next(err);
 };
 
-exports.handle500 = (err,req,res,next)=>{
-  console.log(err)
-  res.status(500).send({msg:'Server Error'})
-}
\ No newline at end of file
+exports.handleBadUser = (err, req, res, next) => {
+  if (err.code === "23503" || err.constraint === "comments_author_fkey") {
+    res.status(404).send({ msg: "Please login or signup before commenting" });
```

Comment:
I would consider making your message more generic here - nothing wrong with it per se - I would at least suggest an `&&` instead as there may be other errors that you come across associated with 23503 that have nothing to do with users for example

Reply:
- [ X ] updated to && , generalised to please log in sign up , can add more to conditional statement when more is implemented